src/Form/EmbargoesIpRangeEntityForm.php

```diff
@@ -82,6 +85,24 @@ public function form(array $form, FormStateInterface $form_state) {
     return $form;
   }
 
+  /**
+   * Validates the IP range entered.
+   *
+   * @param array $element
+   *   An array representing the element.
+   * @param Drupal\Core\Form\FormStateInterface $form_state
+   *   The Drupal form state.
+   */
+  public function validateIpRanges(array $element, FormStateInterface $form_state) {
+    $errors = $this->ipRanges->detectIpRangeStringErrors(array_map('trim', explode('|', trim($form_state->getValue('range')))));
+    if (!empty($errors)) {
+      $form_state->setError($element, $this->t('Problems detected with the %label IP Range.', ['%label' => $this->entity->label()]));
+      foreach ($errors as $error) {
+        $this->messenger()->addError($this->t('Error: %error.', ['%error' => $error]));
```

Comment:
Should... this also just `$form_state->setError()` on the element, instead of dealing separately with messages?

Reply:
Will do this.